sdk/tables/data-tables/samples-dev/generateTableSAS.ts

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+/**
+ * This sample demonstrates how to create an Account SAS token. An account SAS token
+ * provides access to the whole Tables Service account, given the permissions selected,
+ * by default the only permission granted is list
+ *
+ * @summary generate a Table Account SAS token
+ * @azsdk-weight 40
+ */
+
+import {
+  generateAccountSAS,
+  generateTableSAS,
+  TableClient,
+  TableServiceClient,
+  AccountSASPermissions,
+  TableSASPermissions
+} from "@azure/data-tables";
+import { AzureNamedKeyCredential, AzureSASCredential } from "@azure/core-auth";
+
+// Load the .env file if it exists
+import * as dotenv from "dotenv";
+dotenv.config();
+
+const tablesUrl = process.env["TABLES_URL"] || "";
+const accountKey = process.env["ACCOUNT_KEY"] || "";
+const accountName = process.env["ACCOUNT_NAME"] || "";
+
+async function generateTableSASSample() {
+  console.log("== Generate Table Account SAS Sample ==");
+
+  // We need a NamedKeyCredential to generate the SAS token
+  const cred = new AzureNamedKeyCredential(accountName, accountKey);
+
+  // We can optionally set the permissions we want on the SAS token
+  // If non is specified, only list is granted
+  const permissions = new AccountSASPermissions();
+  // Grants permission to list tables
+  permissions.list = true;
```

Comment:
this pattern feels awkward to me, like it could have just been an object like

```ts
const permissions: AccountSasPermissions = {
   list: true,
   ...
}
```